### PR TITLE
Add original error context info to GemRequireError message

### PR DESCRIPTION
**Pull request by RochesterinNYC**
Addresses #4182.

### CODE REVIEW

**Comment by RochesterinNYC** [comment]
Hm, actually, @indirect, any tips on a quick way to reproduce the error condition without requiring something running stuff in a Docker container or uninstalling my node environment? I'd like to check the formatting and see if it's clean looking.

File: lib/bundler/errors.rb [hunk 1]
```diff
@@ -41,7 +41,10 @@ class GemRequireError < BundlerError
     attr_reader :orig_exception
 
     def initialize(orig_exception, msg)
-      super(msg)
+      full_message = msg + "\nError was: #{orig_exception.message}\n"\
+                      "Backtrace for gem load error was:\n"\
+                      "#{orig_exception.backtrace.inspect}"
```
Comment (segiddins) [inline lib/bundler/errors.rb]: Backtrace should be joined rather than inspected.

**Comment by segiddins** [comment]
Build a gem that just raises in the file bundler will attempt to require?

**Comment by RochesterinNYC** [comment]
Got y'all. Updated the PR with a change to the appropriate spec and updated the formatting.
The output will now look like:
```
/path/bundler/lib/bundler/runtime.rb:80:in `rescue in block (2 levels) in require': There was an error while trying to load the gem 'faulty'. (Bundler::GemRequireError)
Gem Load Error is: Gem Internal Error Message
Backtrace for gem load error is:
/path/bundler/tmp/libs/faulty-1.0.0/lib/faulty.rb:1:in `<top (required)>'
/path/bundler/lib/bundler/runtime.rb:77:in `require'
/path/bundler/lib/bundler/runtime.rb:77:in `block (2 levels) in require'
/path/bundler/lib/bundler/runtime.rb:72:in `each'
/path/bundler/lib/bundler/runtime.rb:72:in `block in require'
/path/bundler/lib/bundler/runtime.rb:61:in `each'
/path/bundler/lib/bundler/runtime.rb:61:in `require'
/path/bundler/lib/bundler.rb:100:in `require'
-e:2:in `<main>'
Bundler Error Backtrace:
  from /path/bundler/lib/bundler/runtime.rb:76:in `block (2 levels) in require'
  from /path/bundler/lib/bundler/runtime.rb:72:in `each'
  from /path/bundler/lib/bundler/runtime.rb:72:in `block in require'
  from /path/bundler/lib/bundler/runtime.rb:61:in `each'
  from /path/bundler/lib/bundler/runtime.rb:61:in `require'
  from /path/bundler/lib/bundler.rb:100:in `require'
  from -e:2:in `<main>'
```
The backtrace for the gem load error would, for the `coffee-rails` example that the original issue #4182 brought up, be all stuff in the internals of the coffee-rails gem presumably. It's only still bundler-related code/files in backtrace for this example output because this is copied from the test suite.

**Comment by segiddins** [comment]
👍🏻

**Comment by indirect** [comment]
@homu r+

**Comment by emirkin** [comment]
@RochesterinNYC what is the simplest way to pick up this fix considering that the new version of Bundler won't be released for a while?

**Comment by RochesterinNYC** [comment]
@emirkin a new release of `v1.12.0` will be made very soon.

If you really don't want to wait for it, though, you can install `v1.12.0.pre.2` via `gem install bundler --pre`

**Comment by emirkin** [comment]
@RochesterinNYC Thanks!

**Pull request merged**